src/CompositeRheologies.jl

```diff
@@ -557,6 +716,90 @@ end
     return j
 end
 
+@generated function fill_J_plastic!(J, r, x, c::CompositeRheology{T, N, Npar, is_par, Nplast, is_plastic}, args) where {T, N, Npar, is_par, Nplast, is_plastic}
+    quote
+        Base.@_inline_meta
+        j = 1
+        Base.Cartesian.@nexprs $N i -> j = @inbounds _fill_J_plastic!(J, r, x, c.elements[i], args, $(is_plastic)[i], $(is_par)[i], j)
+        return nothing
+    end
+end
+
+@inline function _fill_J_plastic!(J, r, x, element, args, is_plast, is_par, j)
+    !is_plast && return j
+
+    j       += 1
+    λ̇       = x[j]
+    id_par  = findall(is_par);
```

Comment:
I'm confused by this line, isn't `ispar` just `true` or `false`?

Reply:
yes, `ispar` is indeed `true/false`; how do you suggest to do it instead?

Reply:
was changed meanwhile